dds/DCPS/DiscoveryBase.h

```diff
@@ -1263,6 +1200,253 @@ namespace OpenDDS {
         }
       }
 
+      void
+      remove_expired_endpoints(const DCPS::MonotonicTimePoint& /*now*/)
+      {
+        MatchingDataIter it;
+        for (it = matching_data_buffer.begin(); it != matching_data_buffer.end(); it++) {
+          if (MonotonicTimePoint::now() - it->second.time_added_to_map >= max_reply_period) {
```

Comment:
Use the "now" from the argument?

Reply:
Isn't MonotonicTimePoint::now() a static operation?

Reply:
Or maybe I should ask, what would be the difference of using it from the argument?

Reply:
Do you want to "sample" the system clock again, or use a previous sample?

Reply:
> Do you want to "sample" the system clock again, or use a previous sample?

Yes, that's what I really meant by my question. I don't actually know the answer, I was just wondering if it was considered.

Reply:
My intention was to sample the system clock again, and check if the time expired has exceeded the predefined value